File: oct_2025/oct_1st_2025/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    static void backtrack(vector<int>& nums, vector<vector<int>>& result, vector<int>& path) {
        if (nums.empty()) {
            result.push_back(path);
            return;
        }

        for (int i = 0; i < nums.size(); i++) {
            if (i > 0 && nums[i] == nums[i - 1]) continue; 

            int val = nums[i];
            nums.erase(nums.begin() + i);
            path.push_back(val);

            backtrack(nums, result, path);

            path.pop_back(); 
            nums.insert(nums.begin() + i, val);
        }
    }

    static vector<vector<int>> uniquePerms(vector<int> nums) {
        vector<vector<int>> result;
        vector<int> path;
        sort(nums.begin(), nums.end());
        backtrack(nums, result, path);
        return result;
    }
};

File: oct_2025/oct_1st_2025/Solution.java
import java.util.*;
class Solution {
    static void backtrack(ArrayList<Integer> nums,ArrayList<ArrayList<Integer>> arr, ArrayList<Integer> list){
        if(nums.size()==0){
            arr.add(new ArrayList<>(list));
            return;
        }
        for(int i=0;i<nums.size();i++){
            if(i > 0 && nums.get(i) == nums.get(i-1)) continue;
            int val=nums.get(i);
            nums.remove(i);
            list.add(val);
            backtrack(nums,arr,list);
            list.remove(list.size()-1);
            nums.add(i,val);
        }
    }
    public static ArrayList<ArrayList<Integer>> uniquePerms(int[] nums) {
        ArrayList<ArrayList<Integer>> arr=new ArrayList<>();
        ArrayList<Integer> list=new ArrayList<>();
        for(int i=0;i<nums.length;i++)list.add(nums[i]);
        Collections.sort(list);
        backtrack(list,arr,new ArrayList<>());
        return arr;
    }
};